src/components/common/FavoriteButton.js

```diff
@@ -0,0 +1,73 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import IconButton from '@material-ui/core/IconButton';
+import FavoriteIcon from '@material-ui/icons/Favorite';
+import FavoriteBorderIcon from '@material-ui/icons/FavoriteBorder';
+import { useTranslation } from 'react-i18next';
+import Tooltip from '@material-ui/core/Tooltip';
+import { MUTATION_KEYS } from '@graasp/query-client';
+import { FAVORITE_ITEM_BUTTON_CLASS } from '../../config/selectors';
+import { useMutation } from '../../config/queryClient';
+import { isItemFavorite } from '../../utils/item';
+
+const FavoriteButton = ({ item, member }) => {
+  const { t } = useTranslation();
+  const mutation = useMutation(MUTATION_KEYS.EDIT_MEMBER);
+
+  const [isFavorite, setIsFavorite] = useState(isItemFavorite(item, member));
```

Comment:
Is it necessary to save `isFavorite` in a state? I think you can just compute the value and use it. But maybe I'm missing something.